docs/sovereign/from-sovereign.md

```diff
@@ -0,0 +1,111 @@
+2. The function to call
+3. Any required arguments
+4. *gasLimit* if a smart contract execution is needed after the cross-chain operation
+
+Each action that can be executed remotely through this contract is called an *Operation*. The endpoint responsible for executing those operations is called `execute_operations`.
+
+#### Execution from inside the Sovereign Chain flow
+1. User sends token to the ESDT-Safe smart contract on Sovereign.
+2. The validators generate a proof on the Sovereign Chain for a batch of transfers, which will be sent to the mainchain ESDT-Safe contract.
+3. Validators add this information to the sovereignChainBlockBody. Otherwise the block is not signed.
+4. Leader will push the created txData to the mainchain
+5. The ESDT-Safe contract on the mainchain verifies the proof and executes the transfers.
+6. Sovereign validators notarize the completion of the transfer in the subsequent Sovereign block by receiving the attestation log event directly from the mainchain.
+
+Before talking about the logic of the endpoint, let’s give some more context about the *Operation*.
+
+This structure, derived with multiple traits for serialization, encoding, and type interfacing, represents a fundamental unit of execution within the bridging mechanism.
+
+:::note
+The source code for the following structures can be found [here](https://github.com/multiversx/mx-sovereign-sc/blob/main/common/transaction/src/lib.rs)
+:::
+
+```rust
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, ManagedVecItem, Clone)]
+pub struct Operation<M: ManagedTypeApi> {
+    pub to: ManagedAddress<M>,
+    pub tokens: ManagedVec<M, OperationEsdtPayment<M>>,
+    pub data: OperationData<M>,
+}
+```
+
+- `to`: specifies the destination of the *Operation*
+- `tokens`: represents one or more token transfers associated with the operation
+- `data`: encapsulates additional instructions or parameters that guide the execution of the operation
+
+Why do we need so many custom structures? This a question related to clean code principles, the answer is easy: to ensure that complex logic remains organized and maintainable, thus improving code extension and flexibility.
+
+```rust
+pub struct OperationEsdtPayment<M: ManagedTypeApi> {
+    pub token_identifier: TokenIdentifier<M>,
+    pub token_nonce: u64,
+    pub token_data: EsdtTokenData<M>,
+}
+```
+
+This struct describes a single token transfer action within an Operation. Each Operation can have one or more of such payments, with that enabling the transfer of a variety of tokens during a cross-chain transaction.
+
+- `token_identifier`: used for the identification of the token
+- `token_nonce`: if the token is Non-Fungible or Semi-Fungible, it will have a custom nonce, if not the value will be 0
+- `token_data`: a structure holding metadata and additional token-related details used for minting, token creation (for more details about this structure, please visit the official ESDT Documentation)
```

Comment:
You can use this reference for token data `/developers/developer-reference/sc-api-functions#get_esdt_token_data`